log_collector.py

```diff
@@ -16,6 +16,7 @@
 from collections import OrderedDict
 import shutil
 import json
+import pathlib
```

Comment:
We would prefer to not install any libraries, as this requires the user to install them as well.
If this script becomes more complex to maintain, it is no longer a simple script, and should be deployed using containers.

Reply:
pathlib is part of the standard Python 3 library (cf. https://docs.python.org/3/library/pathlib.html), so there's no need for a user to explicitly install this module.  In contrast to yaml :-).